great_expectations/core/domain_types.py

```diff
@@ -0,0 +1,88 @@
+from enum import Enum, EnumMeta
+
+
+# Reference: "https://stackoverflow.com/questions/33679930/how-to-extend-python-enum".
+class MetaClsEnumJoin(EnumMeta):
+    """
+    Metaclass that creates a new `enum.Enum` from multiple existing Enums.
+
+    @code
+        from enum import Enum
+
+        ENUMA = Enum('ENUMA', {'a': 1, 'b': 2})
+        ENUMB = Enum('ENUMB', {'c': 3, 'd': 4})
+        class ENUMJOINED(Enum, metaclass=MetaClsEnumJoin, enums=(ENUMA, ENUMB)):
+            pass
+
+        print(ENUMJOINED.a)
+        print(ENUMJOINED.b)
+        print(ENUMJOINED.c)
+        print(ENUMJOINED.d)
+    @endcode
+    """
+
+    # noinspection PyMethodParameters
+    @classmethod
+    def __prepare__(metaclass, name, bases, enums=None, **kwargs):
```

Comment:
it is not entirely intuitive what this method does - could we name it something more intuitive / explain it with a better docstring?

Reply:
@gilpasternak35 This is an interface method, and I would rather not disturb it, since it comes straight from the referred article.